worlds/sc2/__init__.py

```diff
@@ -136,223 +174,371 @@ def setup_events(player: int, locked_locations: typing.List[str], location_cache
+    if world.options.spear_of_adun_autonomously_cast_ability_presence == SpearOfAdunAutonomouslyCastAbilityPresence.option_not_present:
+        soa_missions = missions
+        if not world.options.spear_of_adun_autonomously_cast_present_in_no_build:
+            soa_missions = [m for m in soa_missions if MissionFlag.NoBuild not in m.flags]
+        if world.options.spear_of_adun_autonomously_cast_ability_presence == SpearOfAdunAutonomouslyCastAbilityPresence.option_lotv_protoss:
+            soa_missions = [m for m in soa_missions if m.campaign == SC2Campaign.LOTV]
+        elif world.options.spear_of_adun_autonomously_cast_ability_presence == SpearOfAdunAutonomouslyCastAbilityPresence.option_protoss:
+            soa_missions = [m for m in soa_missions if MissionFlag.Protoss in m.flags]
+        
+        soa_passive_presence = len(soa_missions) > 0
+    else:
+        soa_passive_presence = False
+
+    for item in item_list:
+        # Filter Nova equipment if you never get Nova
+        if not nova_missions and (item.name in ItemGroups.nova_equipment):
+            item.flags |= ItemFilterFlags.Excluded
+        
+        # Todo(mm): How should no-build only / grant_story_tech affect excluding Kerrigan items?
+        # Exclude Primal form based on Kerrigan presence or primal form option
+        if (item.data.type == Items.ZergItemType.Primal_Form
+            and ((not kerrigan_is_present) or world.options.kerrigan_primal_status != KerriganPrimalStatus.option_item)
+        ):
+            item.flags |= ItemFilterFlags.Excluded
+        
+        # Remove Kerrigan abilities if there's no kerrigan
+        if (item.data.type == Items.ZergItemType.Ability and not kerrigan_is_present):
+            item.flags |= ItemFilterFlags.Excluded
+        
+        # Remove Spear of Adun if it's off
+        if item.name in Items.spear_of_adun_calldowns and not soa_presence:
+            item.flags |= ItemFilterFlags.Excluded
+
+        # Remove Spear of Adun passives
+        if item.name in Items.spear_of_adun_castable_passives and not soa_passive_presence:
+            item.flags |= ItemFilterFlags.Excluded
+        
+        # Remove Psi Disrupter and Hive Mind Emulator if you never play a build TvZ
+        if (item.name in (ItemNames.HIVE_MIND_EMULATOR, ItemNames.PSI_DISRUPTER)
+            and not tvz_build_missions
+        ):
+            item.flags |= ItemFilterFlags.Excluded
+    return
+
+
+def flag_allowed_orphan_items(world: SC2World, item_list: List[FilterItem]) -> None:
+    """Adds the `Allowed_Orphan` flag to items that shouldn't be filtered with their parents, like combat shield"""
+    missions = get_all_missions(world.mission_req_table)
+    terran_missions = any(MissionFlag.Terran in  mission.flags for mission in missions)
+    zerg_missions = any(MissionFlag.Terran in  mission.flags for mission in missions)
+    protoss_missions = any(MissionFlag.Terran in  mission.flags for mission in missions)
```

Comment:
These should only trigger when terran no-builds are in the pool, when Zergling reconstitution is in the pool, and when specifically Evil Awoken is in the pool, rather than any terran/zerg/protoss missions.